src/Workspaces/Core/Portable/Workspace/Solution/AnalyzerConfigDocumentState.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class AnalyzerConfigDocumentState : TextDocumentState
```

Comment:
1. maybe a comment that this is for .editorconfig.
2. i'm surprised we need a dedicated subclass here.  why can't we just use the normal one?  is this so that this type serves as a green node, and can carry it's data arond with it as long as the contents don't change?

Reply:
i would almost prefer if TextDocumentState were just generic over some piece of computed data it held onto.  but maybe that's just awful in another direction.

Reply:
Correct, there's a dedicated subclass so it's holding onto the existing parsed version. I expect additional logic to land here too, because if somebody wants to write a refactoring that changes a .editorconfig, they might be able to provide the parsed version and then this will like regular syntax trees defer the regeneration of the text.